forem/forem
# Add crayons modal to GDPR confirm delete flow

## What type of PR is this? (check all applicable)


## Description

This PR replaces the "window confirm" (the browser default popup) for marking a GDPR delete request as complete with a crayons modal. I've also added a flash confirmation message on success/error, which we didn't previously have and felt inconsistent with our other member actions.

## Related Tickets & Documents



- Closes https://github.com/forem/forem/issues/17468

## QA Instructions, Screenshots, Recordings

- Log in as an admin
- Visit a user from the admin member index view and delete them (you can pick this from the dropdown)
- Now visit the GDPR delete requests tab, where you should see your deleted user
- Click the 'Deleted' button and check the modal pops up with the correct username
- Confirm the deletion
- Check the modal closes, page reloads, and a flash message appears to confirm success

https://user-images.githubusercontent.com/20773163/166684680-b95ffb72-148e-4e81-a02e-8f3b84a2b2b7.mp4


### UI accessibility concerns?

- Unique button names have been added in this PR, making it easier for screen reader users to confirm deletion of a specific user
- Deletion is now inside a form (unlike before which was a link URL), improving semantic HTML

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] This change does not need to be communicated, and this is why not: it will be shared alongside other member index phase 2 changes


## [optional] What gif best describes this PR or how it makes you feel?

![a bobbing head repeatedly hits the "delete this" key on a keyboard](https://media.giphy.com/media/xULW8N9O5WD32L5052/giphy.gif)
